ibis/backends/base/sql/alchemy/registry.py

```diff
@@ -333,6 +333,22 @@ def _window(t, op):
             'for SQLAlchemy-based backends.'
         )
 
+    # Checks for invalid user input e.g. passing in tuple for preceding and
+    # non-None value for following are caught and raised in expr/window.py
+    # if we're here, then the input is valid, we just need to interpret it
+    # correctly
+    if isinstance(window.preceding, tuple):
+        preceding, following = map(
```

Comment:
Let's call these `start` and `end` or something like that, since `preceding` and `following` are misleading names in the cases of `preceding=(-3, -1)`, which renders as `3 PRECEDING AND 1 PRECEDING`.